File: lib/domain/model/CoffeeDomainModel.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'CoffeeDomainModel.freezed.dart';

@freezed
class CoffeeDomainModel with _$CoffeeHotDomainModel {
  const factory CoffeeDomainModel({
    required String title,
    required String description,
    required List<String> ingredients,
    required String image,
    required int id,
  }) = _CoffeeHotDomainModel;
}

File: app/src/main/java/com/example/coffeepedia/domain/model/CoffeeDomainModel.kt
package com.example.coffeepedia.domain.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class CoffeeDomainModel(
    val title: String,
    val description: String,
    val ingredients: List<String>,
    val image: String,
    val id: String,
) : Parcelable